sdks/python/apache_beam/transforms/fully_qualified_named_transform.py

```diff
@@ -52,16 +52,37 @@ def __init__(self, constructor, args, kwargs):
     self._kwargs = kwargs
 
   def expand(self, pinput):
-    return pinput | self._resolve(self._constructor)(
-        *self._args, **self._kwargs)
+    if self._constructor in ('__callable__', '__constructor__'):
+      self._check_allowed(self._constructor)
+      if self._args:
+        source, *args = tuple(self._args)
+        kwargs = self._kwargs
+      else:
+        args = self._args
+        kwargs = dict(self._kwargs)
+        source = kwargs.pop('source')
+
+      if self._constructor == '__constructor__':
```

Comment:
So in the first case the function returns the PTransform. In the second case the function defines the "expand" method of a PTransform. Is that correct ?

We should document all this properly in JavaExternalTransform since this usage is not clear without going through the details of the code.

Reply:
Fair point. Done.